shared/middleware/helpers/file_bucket.rb

```diff
@@ -13,14 +13,37 @@ def allowed_file_type?(extension)
     Sinatra::Base.mime_type(extension)
   end
 
+  def get_manifest(channel_id)
+    manifest_result = get(channel_id, FileBucket::MANIFEST_FILENAME)
+    if manifest_result[:status] == 'NOT_FOUND'
+      return []
+    end
+    JSON.load manifest_result[:body]
+  end
+
   def copy_files(src_channel, dest_channel, options={})
     # copy everything except the manifest
     options[:exclude_filenames] = [MANIFEST_FILENAME]
     result = super src_channel, dest_channel, options
+    # return right away if there are no files in the source project
+    return if result.empty?
+
+    # update dest_manifest with the decorated filenames from the src_manifest
+    # (we need the new version ids from the dest_manifest, otherwise we'd just copy the src_manifest)
+    dest_manifest = JSON.load result.to_json
```

Comment:
I did this to avoid problems I was seeing with hashes having symbols instead of strings (e.g. `dest_entry['filename']` wouldn't work, but `dest_entry[:filename]` would work). This made the codepath that created the `dest_manifest` look more like the codepath that created the `src_manifest`, even though it seemed a little inefficient.